pkg/handler/enqueue_mapped.go

```diff
@@ -81,7 +83,15 @@ func (e *enqueueRequestsFromMapFunc[object, request]) Create(
 	q workqueue.TypedRateLimitingInterface[request],
 ) {
 	reqs := map[request]empty{}
-	e.mapAndEnqueue(ctx, q, evt.Object, reqs)
+
+	var lowPriority bool
+	if reflect.TypeFor[object]().Implements(reflect.TypeFor[client.Object]()) && isPriorityQueue(q) && !isNil(evt.Object) {
```

Comment:
I know this is probably a minor optimization, but what do you think about storing the result of `reflect.TypeFor[client.Object]()` in a packge-global unexported variable? This reduces the number of reflection calls quite a bit.

(Fine to keep it if you want, maybe the compiler optimizes it away anyway)

Reply:
Hm, could it be that the entire check with reflection here doesn't change during the lifetime of enqueueRequestsFromMapFunc? 

If yes, can we just run this once in TypedEnqueueRequestsFromMapFunc and store the result in enqueueRequestsFromMapFunc?